### ISSUE TITLE
Termux + Catch2 Log "Fix"

### ISSUE THREAD

**Issue by dnllln** [exercism/cpp#488]
I'm using Exercism's CLI I rebuilt so that it would work through Termux and aarch64 on my phone.

Unfortunately, the catch.hpp build included has CATCH_CONFIG_ANDROID_LOGWRITE defined which causes an "undefined reference to `__android_log_write" linker error upon using make to run the included test.

I'm very new to c++, cmake, and compiling in general, but I was able to define logwrite to CATCH_CONFIG_NO_ANDROID_LOGWRITE on line 451 in catch.hpp and that solved my issue. 

I tried building a global version of Catch2, but without sudo/root I couldn't complete the install. I'm not sure what shared library I need for log to work correctly, but my hacky solution seemed to solve the issue and the test completed.

I hope this bit of information helps anyone else with a similar setup, and I'm open to advice!

EDIT: To quickly add, the folder that needs to be symlinked with ls -s for #include <android/log.h> to link correctly is located at $PREFIX/include/android/log.h within Termux. Placing a pointer in $PREFIX/lib/clang/{version}/include/android/ to the aforementioned /android/ works as well without having to edit the catch.hpp file at all.

**Comment by patricksjackson** [comment]
Thanks, I believe that's a pretty niche usecase, but your solution does make sense and as far as I know should work.

The easiest way to do the workaround should be to modify the `*_test.cpp` file and add the following line at the very top, before catch.hpp is included

```
#define CATCH_CONFIG_NO_ANDROID_LOGWRITE
```

I'm closing this, it will still be searchable for anyone running into a similar issue. Thanks!

**Comment by siebenschlaefer** [comment]
Alternatively you could pass the option `-DCATCH_CONFIG_NO_ANDROID_LOGWRITE=1` to `cmake`

**Comment by dnllln** [comment]
I did some more poking around. So the correct include directory for Termux, $PREFIX/include, is detected by cmake, but the library log.h isn't linked for the included catch.hpp file. 

I added target_link_libraries(${exercise} PRIVATE -llog) to line 36 in CMakeLists.txt after the add_executable call which also corrected the issue. 

Again, very niche case. Just wanted to add to for anyone else in Termux having this issue.
